static/js/components/ReplyButton.js

```diff
@@ -0,0 +1,61 @@
+// @flow
+import React from "react"
+
+import LoginPopup from "./LoginPopup"
+import { commentLoginText, userIsAnonymous } from "../lib/util"
+
+import type { BeginEditingFunc } from "./CommentTree"
+
+type Props = {
+  beginEditing?: BeginEditingFunc,
+  formKey: string,
+  initialValue: Object
+}
+
+type State = {
+  popupVisible: boolean
+}
+
+export default class ReplyButton extends React.Component<Props, State> {
+  constructor() {
+    super()
+    this.state = {
+      popupVisible: false
+    }
+  }
+
+  onTogglePopup = async () => {
+    const { popupVisible } = this.state
+    this.setState({
+      popupVisible: !popupVisible
+    })
+  }
+
+  render() {
+    const { beginEditing, formKey, initialValue } = this.props
+    const { popupVisible } = this.state
+    return (
+      <div>
+        <div
+          className="comment-action-button reply-button"
+          onClick={e => {
+            if (beginEditing && !userIsAnonymous()) {
+              beginEditing(formKey, initialValue, e)
+            } else {
+              this.onTogglePopup()
+            }
+          }}
```

Comment:
I think here we might want to do something like

```js
onClick={
  beginEditing && !userIsAnonymous()
    ? beginEditing(formKey, initialvalue)
    : this.onTogglePopup
}
```

then we can add an `e.preventDefault` in `this.onTogglePopup` to make sure we handle that.